### DISCUSSION TITLE
July 9 - Literate Programming III - testing & notebooks

### DISCUSSION THREAD

**Discussion by tonyfast** [Quansight/writers-workshop#21] [New class]
# the goal of the workshop

[frame notebooks as tests and consider how they can provide value longer.](https://gist.github.com/tonyfast/e7dd7ff3d808d57b77e9765626a73a91)

## so far

1. introduction to literate programs where code and narrative contribute equally to the narrative.
2. introduction to different media formats in the notebook and their use as collage.

## teacher's preferred ground rules

* How to use the raise handle feature? Keeping the professor on track..
* I encourage y'all to interrupt me when I'm talking or demo'ing. 
  Sometimes I get distracted by my own special interests and this class if for you all.
* To help take notes, please respond to this discussion item and continue editing your one reply during class. i can see the changes on the discussion board as they happen.

## [calls to action] :+1: :-1:  :smile: :confused:  :rocket:  :heart:  :eyes: :tada: attendance

* place any questions you have in the discussions and we can try to discuss them during the workshop.

* react to this discussion so we can take attendance.

![image](https://user-images.githubusercontent.com/4236275/117461813-fd163180-af1b-11eb-901b-bfdee65d8bdc.png)

### Q1 how's your energy

* :+1: i feel energetic from this week
* :-1: i feel exhausted from this week

## note taking

please use this discussion as a place to take notes. create a comment that you edit throughout the lesson. 

_the perfect is the enemy of the good_

## agenda

0. [:00-:07] chit chat
1. show and tell
2. testing in notebooks
3. what are some ways we can thing of using notebooks outside of the interactive context?
4. homeworks
5. [:50-:60] wrapup

write! right?

**Comment by MarsBarLee** [comment]
Notes:

- A testing unit should focus on tiny bit of functionality and prove it correct]
- 5 Vs of Big Data -> Veracity, Value are uncountable -> count Veracity, Value in notebooks 
- Why is veracity important in notebooks? Need to replicate results in science
- Structuralism: separation of signifier (eg what the author thinks the word means) and the signified (eg audience meaning) -> breaking our assumption of our writing because language isn't still
- Challenging structuralism in literate programming by adding code to written language
- "We can make the value of our work if we can find ways to make our writing valuable at rest" -> my thinking- notebook still legible and useful even when printed? and then put on a paper collage?
- nbval: test input of notebook to output with pytest results
- testing from beginning/test-forward appraoch: make notebooks into live documentation
- impact of science on language?
         - mimeme -> posting memes
         - virus -> viral videos
         - the common-usage of 'theory' vs 'meaning of 'theory' in natural science 

Homework: write a test in jupyter notebook
